designer_v2/lib/router.dart

```diff
@@ -1,82 +1,152 @@
+import 'package:studyu_designer_v2/common_views/pages/splash_page.dart';
+import 'package:studyu_designer_v2/repositories/auth_repository.dart';
+import 'package:studyu_designer_v2/utils/combined_stream_notifier.dart';
+import 'package:studyu_designer_v2/utils/debug_print.dart';
 
-// List of all pages in the application
-// To create a new page, 1) add an entry to the enum and 2) add the page to the AppRouter class
+/// To create a new page:
+/// 1) add an entry to the [RouterPage] enum
+/// 2) add the page widget to the [routes] list
+
+/// List of all pages in the application
 enum RouterPage {
-  dashboard(title: "dashboard", path: "/"),
-  splash(title: "splash", path: "/splash"),
-  login(title: "login", path: "/login"),
-  error(title: "error", path: "/error");
+  dashboard(id: "dashboard", path: "/studies"),
+  dashboardOwned(id: "dashboardOwned", path: "owned"),
+  dashboardShared(id: "dashboardShared", path: "shared"),
+  studyRegistry(id: "studyRegistry", path: "registry"),
+  splash(id: "splash", path: "/splash"),
+  login(id: "login", path: "/login"),
+  error(id: "error", path: "/error");
 
-  final String title;
+  final String id;
   final String path;
 
-  const RouterPage({required this.title, required this.path, });
+  const RouterPage({required this.id, required this.path});
 }
 
-// Wrapper around GoRouter for injecting dependencies such as auth services,
-// allowing for dynamic redirecting based on auth status managed in dedicated
-// state providers
-class AppRouter {
-  late final AppDelegate appDelegate;
-
-  GoRouter get router => _goRouter;
-
-  AppRouter({required this.appDelegate});
-
-  late final GoRouter _goRouter = GoRouter(
-    refreshListenable: appDelegate,
-    initialLocation: RouterPage.dashboard.path,
-    routes: <GoRoute>[
+/// Each route defines a mapping between the URL path and Flutter widget
+final List<GoRoute> routes = [
+  GoRoute(
+    path: RouterPage.dashboard.path,
+    name: RouterPage.dashboard.id,
+    builder: (context, state) => DashboardScreen(),
+    routes: [
       GoRoute(
-        path: RouterPage.dashboard.path,
-        name: RouterPage.dashboard.title,
-        builder: (context, state) => const StudyDashboardScreen(),
+        path: RouterPage.dashboardOwned.path,
+        name: RouterPage.dashboardOwned.id,
+        builder: (context, state) => DashboardScreen(),
       ),
       GoRoute(
-        path: RouterPage.splash.path,
-        name: RouterPage.splash.title,
-        builder: (context, state) => const SplashPage(),
+        path: RouterPage.dashboardShared.path,
+        name: RouterPage.dashboardShared.id,
+        builder: (context, state) => DashboardScreen(),
       ),
       GoRoute(
-        path: RouterPage.login.path,
-        name: RouterPage.login.title,
-        builder: (context, state) => LoginPage(),
+        path: RouterPage.studyRegistry.path,
+        name: RouterPage.studyRegistry.id,
+        builder: (context, state) => DashboardScreen(),
```

Comment:
Same as above. Would be great to have some kind of hierachy to redirect everything part of the dashboard to `DashboardScreen()`

Reply:
It's not clear to me what problem you are describing here.

Is it the duplication of three different routes all rendering the same widget? Unless you want to write a custom parser to enable more concise route configuration, there's not really a way around that because every route & the corresponding widget need to be registered at some point.